tests/jobserver/views/test_admin.py

```diff
@@ -0,0 +1,99 @@
+import pytest
+from django.contrib.messages.storage.fallback import FallbackStorage
+from django.urls import reverse
+
+from jobserver.models import Backend, User
+from jobserver.views.admin import ApproveUsers
+
+from ...factories import OrgFactory, UserFactory
+
+
+@pytest.mark.django_db
+def test_approveusers_as_non_superuser(rf):
+    request = rf.get("/")
+    request.user = UserFactory()
+
+    response = ApproveUsers.as_view()(request)
+
+    assert response.status_code == 403
+
+
+@pytest.mark.django_db
+def test_approveusers_get_success(rf, superuser):
+    request = rf.get(f"/?user={superuser.pk}")
+    request.user = superuser
+
+    response = ApproveUsers.as_view()(request)
+
+    assert response.status_code == 200
+
+    expected = set(User.objects.filter(pk=superuser.pk))
+    output = set(response.context_data["users"])
+    assert output == expected
+
+
+@pytest.mark.django_db
+def test_approveusers_post_success(rf, superuser):
+    user1 = UserFactory()
+    user2 = UserFactory()
+
+    org = OrgFactory()
+
+    test = Backend.objects.get(name="test")
+    tpp = Backend.objects.get(name="tpp")
+
+    data = {
+        "backends": [test.pk, tpp.pk],
+        "orgs": [org.pk],
+    }
+    request = rf.post(f"/?user={user1.pk}&user={user2.pk}", data)
+    request.user = superuser
+
+    # set up messages framework
+    request.session = "session"
+    request._messages = FallbackStorage(request)
+
+    response = ApproveUsers.as_view()(request)
+
+    assert response.status_code == 302
+    assert response.url == reverse("admin:jobserver_user_changelist")
+
+    user1.refresh_from_db()
+    assert set(user1.backends.all()) == {test, tpp}
+    assert org in user1.orgs.all()
+
+    user2.refresh_from_db()
+    assert set(user2.backends.all()) == {test, tpp}
+    assert org in user2.orgs.all()
```

Comment:
Again, you could beef up this test by:

* only approving users for one backend
* creating a second organisation, but not approving users for it

and then checking that the users are only approved for the expected backend and organisation.

Reply:
Backends already exist because they get created by migrations but definitely good call on the orgs.